feat(data-scrubbing): Add sction about advanced safe field usage (#8405)
getsentry/sentry-docs: src/docs/product/data-management-settings/scrubbing/server-side-scrubbing/index.mdx

- While you cannot delete a single event, you can delete the issue the event is in by clicking the trash icon on the **Issue Details** page.
- If a large number of unique events - representing multiple issues - contain the data, you may need to delete and re-create the project to effectively cleanse the system.

## Safe Fields

> Safe Fields are used to exclude fields and data from scrubbing. For example this event payload:

Reviewer: ```suggestion
Safe Fields are used to exclude fields and data from scrubbing. For example, take the following event payload:
```

```json
{
  "user": {
    "id": "4111111111111111"
  },
  "extra": {
    "id": "sensitive",
    "sys.argv": ["script.py", "password"],
    "credentials": { "username": "jane", "password": "p4ssw0rd!" }
  }
}
```

> With scrubbing enabled and `id` added as a sensitive field, the `user.id` field will be scrubbed
> (it matches a credit card), the `password` string in `extra.'sys.argv'` as well as the entire `credentials` object.

Reviewer 2: ```suggestion
When scrubbing is enabled and `id` added as a sensitive field, the `password` string in `extra.'sys.argv'`, the `user.id` field (because it matches a credit card), as well as the entire `credentials` object will be scrubbed.
```

> To accomodate more fine grained filtering the Safe Fields support the same syntax as the

Reviewer 2: ```suggestion
To accomodate more fine-grained filtering, the Safe Fields support the same syntax as the
```

[Advanced Datascrubbing Sources](/product/data-management-settings/scrubbing/advanced-datascrubbing/#sources).

> To leave the the `user.id` field intact one could configure simply `id` as a Safe Field, but this also
> marks `extra.id` as a Safe Field. In order to only select the user id the full path can be used instead: `user.id`.

Reviewer 2: ```suggestion
If you want to leave the `user.id` field intact, you can configure the `id` as a Safe Field, but this will also
mark `extra.id` as a Safe Field. To specifically select the user id, use the full path instead of `user.id`.
```

> Marking elements of an array as safe requires a wildcard selector, for example `'sys.argv'.*`. A deep wildcard
> can be used to mark all elements contained within an object as safe. The two selectors `extra.credentials`,
> `extra.credentials.**` mark the `credentials` object and all of its members as safe.

Reviewer 2: ```suggestion
Marking elements of an array as safe requires a wildcard selector, for example `'sys.argv'.*`. A deep wildcard
can be used to mark all elements contained within an object. The two selectors: `extra.credentials` and
`extra.credentials**` will mark the `credentials` object and all of its members as safe.
```

> `extra.credentials.**` will mark the `credentials` object and all of its members as safe.

Reviewer 3: I personally would link to https://docs.sentry.io/product/data-management-settings/scrubbing/advanced-datascrubbing/#wildcards instead.
